lib/dal/src/cached_module.rs

```diff
@@ -561,143 +522,199 @@ impl CachedModule {
+            WHERE latest_hash = $2
+        ";
         info!(
-            "Updating sdf module cache for {} - {schema_name} ({category:?})",
-            module_details.name
+            "module {} ({}): {}",
+            module_id,
+            summary.schema_name().unwrap_or("<no name in package>"),
+            summary.package_summary()
         );
-
-        let bytes_ref = pkg_bytes.as_slice();
-        let row = ctx
-            .txns()
+        ctx.txns()
             .await?
             .pg()
-            .query_one(
-                query,
-                &[
-                    &schema_id,
-                    &schema_name,
-                    &display_name,
-                    &category,
-                    &link,
-                    &color,
-                    &description,
-                    &component_type.to_string(),
-                    &module_details.latest_hash,
-                    &module_details.created_at,
-                    &bytes_ref,
-                ],
-            )
+            .execute(query, &[&summary.package_summary(), &hash])
             .await?;
-
-        Ok(Some(row.try_into()?))
+        Ok(())
     }
 
-    pub async fn remove_all_for_schema_id(
+    async fn package_data_for_hash(
         ctx: &DalContext,
-        schema_id: SchemaId,
-    ) -> CachedModuleResult<()> {
-        let query = r#"
-            DELETE FROM cached_modules WHERE schema_id = $1
-            "#;
+        hash: &str,
+    ) -> CachedModuleResult<Option<Vec<u8>>> {
+        let query = "
+            SELECT DISTINCT package_data
+            FROM cached_modules
+            WHERE latest_hash = $1 AND package_data IS NOT NULL
+            LIMIT 1
+        ";
+        let Some(matching_data) = ctx.txns().await?.pg().query_opt(query, &[&hash]).await? else {
+            return Ok(None);
+        };
+        Ok(matching_data.try_get("package_data")?)
+    }
+}
 
-        ctx.txns().await?.pg().query(query, &[&schema_id]).await?;
+struct PackageData {
+    schema: Option<SiPkgSchemaData>,
+    variant: Option<SiPkgSchemaVariantData>,
+    package_summary: serde_json::Value,
+}
 
-        Ok(())
+impl PackageData {
+    async fn load(
+        module_id: &str, // just for debug messages so we can find the broken rows
+        pkg_bytes: &Arc<Vec<u8>>,
+    ) -> CachedModuleResult<Option<Self>> {
+        let pkg_bytes = pkg_bytes.clone();
+        let pkg = slow_rt::spawn(async move { SiPkg::load_from_bytes(&pkg_bytes) })?.await??;
+
+        let Some(schema) = pkg.schemas()?.into_iter().next() else {
+            warn!("builtin module {} has no schema", module_id);
+            return Ok(None);
+        };
+
+        let Some(variant) = schema.variants()?.into_iter().next() else {
+            warn!("builtin module {} has a schema with no variant", module_id);
+            return Ok(None);
+        };
+
+        let package_summary = PackageSummary {
+            socket_count: variant.sockets()?.len() as u32
```

Comment:
This is the calculation. It yields the same result as the existing socket count algorithm that runs against installed packages.